src/awx_plugins/credentials/azure_kv.py

```diff
@@ -71,24 +88,52 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
-    kv = SecretClient(credential=csc, vault_url=kwargs['url'])
-    return kv.get_secret(
-        name=kwargs['secret_field'],
-        version=kwargs.get('secret_version', ''),
+def azure_keyvault_backend(
+        url: str,
+        client: str,
+        secret: str,
+        tenant: str,
+        secret_field: str,
+        secret_version: str,
+) -> str:
+    """Get a credential and retrieve a secret from an Azure Key Vault.
+
+    :param url: An Azure Key Vault URI.
+    :param client: The Client ID  (optional).
```

Comment:
What does “optional” mean here? Can it be `None`?

Reply:
if a Client ID is not provided I believe an empty string will be submitted and a Client ID is not required

Reply:
Then I'd document that “empty string counts as not provided” or something. “Optional” is often interpreted as `None` since it's a popular sentinel value, but in this case it sounds like an empty string is a sentinel.